presto-cassandra/src/main/java/com/facebook/presto/cassandra/CassandraRecordCursor.java

```diff
@@ -111,6 +111,8 @@ public long getLong(int i)
                 return currentRow.getLong(i);
             case TIMESTAMP:
                 return currentRow.getTimestamp(i).getTime();
+            case DATE:
```

Comment:
Like for the other types, you probably need to update CassandraRecordSink for CTAS.